pkg/init/cmd/service/runc.go

```diff
@@ -39,6 +41,10 @@ func runcInit(rootPath string) int {
 
 	status := 0
 
+	if err := os.MkdirAll(logDir, 0755); err != nil {
+		log.Fatalf("Cannot create log directory %s: %v", logDir, err)
```

Comment:
generally we try not to fatally error in init as it is hard to debug. Maybe just print an error and continue on?

Reply:
Although I guess a lot will go wrong below...

Reply:
Crap. Stupid copy-paste error from my tests. Sorry.

Reply:
I take it back. I was thinking of a different part of the file. We use `log.Fatalf` in a number of places in this file, if we think it cannot work.

```go
	// get files; note ReadDir already sorts them
	files, err := ioutil.ReadDir(rootPath)
	if err != nil {
		log.Fatalf("Cannot read files in %s: %v", rootPath, err)
	}

	tmpdir, err := ioutil.TempDir("", filepath.Base(rootPath))
	if err != nil {
		log.Fatalf("Cannot create temporary directory: %v", err)
	}

	// need to set ourselves as a child subreaper or we cannot wait for runc as reparents to init
	if err := sys.SetSubreaper(1); err != nil {
		log.Fatalf("Cannot set as subreaper: %v", err)
	}

	status := 0
```

We fatal out if we cannot:

* read onboot config files
* create temp dir
* set subreaper


I think being unable to write logs is equivalent. Leave it?

Reply:
well if we cant read the config files we cant try to run them. If we can not insanely just fallback to stdout it might be nice, in case one of your onboots leaves `/var` in an unwriteable state temporarily. But not too bothered really, maybe it is better if you fix up the errors.

Reply:
Yeah, I lean towards, "if your onboot screwed up `/var`, I would rather you fail cleanly than get into a half-there not-really-known state."